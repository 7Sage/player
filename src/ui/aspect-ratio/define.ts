import { safelyDefineCustomElement } from '../../utils/dom';
import {
  ASPECT_RATIO_ELEMENT_TAG_NAME,
  AspectRatioElement
} from './AspectRatioElement';

safelyDefineCustomElement(ASPECT_RATIO_ELEMENT_TAG_NAME, AspectRatioElement);

declare global {
  interface HTMLElementTagNameMap {
    [ASPECT_RATIO_ELEMENT_TAG_NAME]: AspectRatioElement;
  }
}
