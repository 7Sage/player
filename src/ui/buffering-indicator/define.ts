import { safelyDefineCustomElement } from '@utils/dom';

import {
  BUFFERING_INDICATOR_ELEMENT_TAG_NAME,
  BufferingIndicatorElement
} from './BufferingIndicatorElement';

safelyDefineCustomElement(
  BUFFERING_INDICATOR_ELEMENT_TAG_NAME,
  BufferingIndicatorElement
);

declare global {
  interface HTMLElementTagNameMap {
    [BUFFERING_INDICATOR_ELEMENT_TAG_NAME]: BufferingIndicatorElement;
  }
}
