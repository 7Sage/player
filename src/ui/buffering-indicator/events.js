import { VdsCustomEvent } from '../../foundation/events/index.js';

export class BufferingIndicatorEvent extends VdsCustomEvent {}

export class BufferingIndicatorShowEvent extends BufferingIndicatorEvent {
  static TYPE = 'vds-buffering-indicator-show';
  constructor(eventInit) {
    super(BufferingIndicatorShowEvent.TYPE, eventInit);
  }
}

export class BufferingIndicatorHideEvent extends BufferingIndicatorEvent {
  static TYPE = 'vds-buffering-indicator-hide';
  constructor(eventInit) {
    super(BufferingIndicatorHideEvent.TYPE, eventInit);
  }
}
