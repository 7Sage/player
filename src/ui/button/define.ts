import { safelyDefineCustomElement } from '../../utils/dom';
import { BUTTON_ELEMENT_TAG_NAME, ButtonElement } from './ButtonElement';

safelyDefineCustomElement(BUTTON_ELEMENT_TAG_NAME, ButtonElement);

declare global {
  interface HTMLElementTagNameMap {
    [BUTTON_ELEMENT_TAG_NAME]: ButtonElement;
  }
}
