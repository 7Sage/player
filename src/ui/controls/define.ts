import { safelyDefineCustomElement } from '@utils/dom';

import { CONTROLS_ELEMENT_TAG_NAME, ControlsElement } from './ControlsElement';

safelyDefineCustomElement(CONTROLS_ELEMENT_TAG_NAME, ControlsElement);

declare global {
  interface HTMLElementTagNameMap {
    [CONTROLS_ELEMENT_TAG_NAME]: ControlsElement;
  }
}
