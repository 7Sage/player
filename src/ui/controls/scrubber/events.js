import { VdsCustomEvent } from '../../../foundation/events/index.js';

/**
 * @typedef {{
 *  [ScrubberPreviewShowEvent.TYPE]: ScrubberPreviewShowEvent;
 *  [ScrubberPreviewHideEvent.TYPE]: ScrubberPreviewHideEvent;
 *  [ScrubberPreviewTimeUpdateEvent.TYPE]: ScrubberPreviewTimeUpdateEvent;
 * }} ScrubberEvents
 */

/**
 * @template DetailType
 * @extends {VdsCustomEvent<DetailType>}
 */
export class ScrubberEvent extends VdsCustomEvent {}

/**
 * Emitted when the preview transitions from hidden to showing.
 *
 * @extends {ScrubberEvent<void>}
 */
export class ScrubberPreviewShowEvent extends ScrubberEvent {
  /** @readonly */
  static TYPE = 'vds-scrubber-preview-show';
}

/**
 * Emitted when the preview transitions from showing to hidden.
 *
 * @extends {ScrubberEvent<void>}
 */
export class ScrubberPreviewHideEvent extends ScrubberEvent {
  /** @readonly */
  static TYPE = 'vds-scrubber-preview-hide';
}

/**
 * Emitted when the time being previewed changes.
 *
 * @extends {ScrubberEvent<number>}
 */
export class ScrubberPreviewTimeUpdateEvent extends ScrubberEvent {
  /** @readonly */
  static TYPE = 'vds-scrubber-preview-time-update';
}
