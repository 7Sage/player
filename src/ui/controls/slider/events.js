import { VdsCustomEvent } from '../../../shared/events/index.js';

export class SliderEvent extends VdsCustomEvent {}

export class SliderValueChangeEvent extends SliderEvent {
  static TYPE = 'vds-slider-value-change';
  constructor(eventInit) {
    super(SliderValueChangeEvent.TYPE, eventInit);
  }
}

export class SliderDragStartEvent extends SliderEvent {
  static TYPE = 'vds-slider-drag-start';
  constructor(eventInit) {
    super(SliderDragStartEvent.TYPE, eventInit);
  }
}

export class SliderDragEndEvent extends SliderEvent {
  static TYPE = 'vds-slider-drag-end';
  constructor(eventInit) {
    super(SliderDragEndEvent.TYPE, eventInit);
  }
}
