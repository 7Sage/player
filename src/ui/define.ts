import './buffering-indicator/define';
import './controls/define';
import './fullscreen-button/define';
import './mute-button/define';
import './play-button/define';
import './scrim/define';
import './scrubber/define';
import './scrubber-preview/define';
import './scrubber-preview-time/define';
import './scrubber-preview-video/define';
import './seekable-progress-bar/define';
import './slider/define';
import './time/define';
import './time-current/define';
import './time-duration/define';
import './time-slider/define';
import './toggle-button/define';
import './volume-slider/define';

export * from '.';
