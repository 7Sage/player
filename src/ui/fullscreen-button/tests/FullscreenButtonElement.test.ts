import '../../../define/vds-fullscreen-button';

import { elementUpdated } from '@open-wc/testing-helpers';
import { html } from 'lit';

import { buildMediaPlayerFixture } from '../../../media/test-utils';
import { waitForEvent } from '../../../utils/events';

async function buildFixture() {
  const { player } = await buildMediaPlayerFixture(html`
    <vds-fullscreen-button>
      <div class="enter"></div>
      <div class="exit"></div>
    </vds-fullscreen-button>
  `);

  const button = player.querySelector('vds-fullscreen-button')!;

  return { player, button };
}

test('light DOM snapshot', async function () {
  const { button } = await buildFixture();
  expect(button).dom.toMatchSnapshot();
});

test('shadow DOM snapshot', async function () {
  const { button } = await buildFixture();
  expect(button).shadowDom.to.equal(`<slot></slot>`);
});

test('it should update `hidden` attribute based on fullscreen support', async () => {
  const { player, button } = await buildFixture();

  player._mediaStore.canFullscreen.set(true);
  await elementUpdated(button);
  expect(button.hasAttribute('hidden')).to.be.false;

  player._mediaStore.canFullscreen.set(false);
  await elementUpdated(button);
  expect(button.hasAttribute('hidden')).to.be.true;
});

test('it should update fullscreen state', async () => {
  const { player, button } = await buildFixture();

  player._mediaStore.fullscreen.set(true);
  await elementUpdated(button);

  expect(button.isPressed).to.be.true;
  expect(button.getAttribute('aria-pressed')).to.equal('true');
  expect(button.hasAttribute('media-fullscreen')).to.be.true;

  player._mediaStore.fullscreen.set(false);
  await elementUpdated(button);

  expect(button.isPressed).to.be.false;
  expect(button.getAttribute('aria-pressed')).to.equal('false');
  expect(button.hasAttribute('media-fullscreen')).to.be.false;
});

test('it should enter fullscreen', async function () {
  const { player, button } = await buildFixture();

  const requestFullscreenSpy = vi
    .spyOn(player, 'requestFullscreen')
    .mockImplementation(() => Promise.resolve());

  player._mediaStore.fullscreen.set(false);
  await elementUpdated(button);

  setTimeout(() => button.dispatchEvent(new MouseEvent('pointerdown')));

  await waitForEvent(button, 'vds-enter-fullscreen-request');
  expect(requestFullscreenSpy).to.toHaveBeenCalledOnce();
});

test('it should exit fullscreen', async function () {
  const { player, button } = await buildFixture();

  const exitFullscreenSpy = vi
    .spyOn(player, 'exitFullscreen')
    .mockImplementation(() => Promise.resolve());

  player._mediaStore.fullscreen.set(true);
  await elementUpdated(button);

  setTimeout(() => button.dispatchEvent(new MouseEvent('pointerdown')));

  await waitForEvent(button, 'vds-exit-fullscreen-request');
  expect(exitFullscreenSpy).to.toHaveBeenCalledOnce();
});
