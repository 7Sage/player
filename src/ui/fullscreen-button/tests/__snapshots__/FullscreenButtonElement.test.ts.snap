// Vitest Snapshot v1

exports[`light DOM snapshot 1`] = `
<vds-fullscreen-button
  aria-label="Fullscreen"
  aria-pressed="false"
  hidden=""
  role="button"
  tabindex="0"
>
  
      
  <div
    class="enter"
  />
  
      
  <div
    class="exit"
  />
  
    
</vds-fullscreen-button>
`;
